opensrp-giz-malawi/src/main/java/org/smartregister/giz/activity/FragmentBaseActivity.java

```diff
@@ -0,0 +1,279 @@
+package org.smartregister.giz.activity;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.PorterDuff;
+import android.graphics.drawable.Drawable;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.Environment;
+import android.view.Menu;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.ActionBar;
+import androidx.appcompat.widget.Toolbar;
+import androidx.core.content.FileProvider;
+import androidx.fragment.app.Fragment;
+
+import org.apache.commons.lang3.StringUtils;
+import org.smartregister.giz.R;
+import org.smartregister.giz.fragment.EligibleChildrenReportFragment;
+import org.smartregister.giz.fragment.FilterReportFragment;
+import org.smartregister.giz.fragment.VillageDoseReportFragment;
+import org.smartregister.giz.listener.PdfGeneratorListener;
+import org.smartregister.giz.util.GizConstants;
+import org.smartregister.giz.util.PdfGeneratorHelper;
+import org.smartregister.util.PermissionUtils;
+import org.smartregister.view.activity.SecuredActivity;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+
+import timber.log.Timber;
+
+public class FragmentBaseActivity extends SecuredActivity {
```

Comment:
This class needs presenters and Ineractors

Reply:
Done

Reply:
Done